src/aliceVision/hdr/brackets.cpp

```diff
@@ -101,18 +103,50 @@ bool estimateBracketsFromSfmData(std::vector<std::vector<std::shared_ptr<sfmData
     return true;
 }
 
-void selectTargetViews(std::vector<std::shared_ptr<sfmData::View>> & out_targetViews, const std::vector<std::vector<std::shared_ptr<sfmData::View>>> & groups, int offsetRefBracketIndex)
+void selectTargetViews(std::vector<std::shared_ptr<sfmData::View>> & out_targetViews, const std::vector<std::vector<std::shared_ptr<sfmData::View>>> & groups, int offsetRefBracketIndex, const std::string& targetIndexesFilename)
 {
-    for(auto & group : groups)
+    // If targetIndexesFilename cannot be opened or is not valid target views are derived from the middle exposed views
+    // For odd number, there is no ambiguity on the middle image.
+    // For even number, we arbitrarily choose the more exposed view (as we usually have more under-exposed images than over-exposed).
+    const int viewNumberPerGroup = groups[0].size();
+    const int middleIndex = viewNumberPerGroup / 2;
+    int targetIndex = middleIndex + offsetRefBracketIndex;
+
+    out_targetViews.clear();
+
+    if ((targetIndex >= 0) && (targetIndex < viewNumberPerGroup))
     {
-        // Target views are the middle exposed views
-        // For add number, there is no ambiguity on the middle image.
-        // For even number, we arbitrarily choose the more exposed view (as we usually have more under-exposed images than over-exposed).
-        const int middleIndex = int(group.size()) / 2;
-        const int targetIndex = clamp(middleIndex + offsetRefBracketIndex, 0, int(group.size()) - 1);
+        ALICEVISION_LOG_INFO("Use offsetRefBracketIndex parameter");
+    }
+    else // try to use the index in the file
+    {
+        ALICEVISION_LOG_INFO("offsetRefBracketIndex parameter out of range, read file containing the target index");
+        std::ifstream file(targetIndexesFilename);
+        if (!file)
+        {
+            return; 
+        }
+        else
+        {
+            std::string line;
+            if (getline(file, line))
+            {
+                targetIndex = atoi(line.c_str());
+                file.close();
+            }
+            else
+            {
+                return;
+            }
+        }
+    }
```

Comment:
```suggestion
        if (!file)
            ALICEVISION_THROW_ERROR("Failed to open file: " << targetIndexesFilename);
        std::string line;
        if (!std::getline(file, line))
            ALICEVISION_THROW_ERROR("Failed to read from file: " << targetIndexesFilename);
        targetIndex = atoi(line.c_str());
    }
```

No need for close: "Note that any open file is automatically closed when the ifstream object is destroyed."